src/app.rs

```diff
@@ -11,101 +11,144 @@ use cargo::util::interning::InternedString;
 use termcolor::Color;
 
 use crate::instruments;
-use crate::opt::{CargoOpts, Opts, Target};
-
-/// The main entrance point, once args have been parsed.
-pub(crate) fn run(args: Opts) -> Result<()> {
-    use cargo::util::important_paths::find_root_manifest_for_wd;
-    instruments::check_existence()?;
-
-    if args.list {
-        let list = instruments::list()?;
-        println!("{}", list);
+use crate::opt::{AppConfig, CargoOpts, Target};
+
+/// Main entrance point, after args have been parsed.
+///
+pub(crate) fn run(app_config: AppConfig) -> Result<()> {
+    // 1. Detect the type of Xcode Instruments installation
+    let xctrace_tool = instruments::XcodeInstruments::detect()?;
+
+    // 2. Render available templates if the user asked
+    if app_config.list_templates {
+        let catalog = xctrace_tool.available_templates()?;
+        println!("{}", instruments::render_template_catalog(&catalog));
         return Ok(());
     }
 
-    let cfg = Config::default()?;
-    let manifest_path = find_root_manifest_for_wd(cfg.cwd())?;
-    let workspace = Workspace::new(&manifest_path, &cfg)?;
-    let workspace_root = manifest_path.parent().unwrap().to_owned();
+    // 3. Build the specified target
+    use cargo::util::important_paths::find_root_manifest_for_wd;
```

Comment:
I would have `use cargo::util::important_paths` at the top of the file, and then call `important_paths::find_root_manifest_for_wd` below.